dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ConfigCenterConfig.java

```diff
@@ -183,13 +213,16 @@ public void setAppConfigFile(String appConfigFile) {
         this.appConfigFile = appConfigFile;
     }
 
-    @Parameter(key = Constants.CONFIG_APPNAME_KEY, useKeyAsProperty = false)
-    public String getAppName() {
-        return appName;
+    public ApplicationConfig getApplication() {
+        if (application != null && !application.isValid()) {
+            throw new IllegalStateException("Name is required for ApplicationConfig, for example, in XML it looks like <dubbo:application name=\"...\" />.");
+        }
+        return application;
     }
 
-    public void setAppName(String appName) {
```

Comment:
is this was public method, will there be any issue with backward compatibility?

Reply:
I am afraid it will break the backward compatibility in theory.
But I tend to fix the inappropriate design directly as there're still few users depends on it.